src/fsharp/IlxGen.fs

```diff
@@ -5089,7 +5089,6 @@ and GenDecisionTreeSwitch cenv cgbuf inplabOpt stackAtTargets eenv e cases defau
             | _ -> error(InternalError("these matches should never be needed", switchm))
 
 and GenDecisionTreeCases cenv cgbuf stackAtTargets eenv defaultTargetOpt targets repeatSP targetInfos sequel caseLabels cases (contf: Zmap<_,_> -> FakeUnit) =
-    assert(cgbuf.GetCurrentStack() = stackAtTargets) // cgbuf stack should be unchanged over tests. [bug://1750].
```

Comment:
Why is this an incorrect assert? I assume based on the comment that it is very old

Reply:
My understanding is it doesn't properly take into some logic associated with "unit" values - the two views of the stack are different.